src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts

```diff
@@ -515,6 +522,23 @@ class SavePlanFile extends Action {
 	}
 
 	public override async run(context: QueryPlan2): Promise<void> {
+		const workspaceFolders = await context.workspaceContextService.getWorkspace().folders;
+		let currentWorkSpaceFolder;
+		if (workspaceFolders.length !== 0) {
+			currentWorkSpaceFolder = workspaceFolders[0].uri;
+		}
+		const saveFileUri = await context.fileDialogService.showSaveDialog({
+			filters: [
+				{
+					extensions: ['sqlplan'], //TODO: Get this extension from provider
+					name: localize('queryPlan.SaveFileDescription', 'Execution Plan Files')
```

Comment:
this should also come from extension in the future, right?

Reply:
Yes, both the extension and name should come from provider